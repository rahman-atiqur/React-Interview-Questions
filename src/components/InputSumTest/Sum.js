import React, { useState } from "react";

const Sum = () => {
    const [inputs, setInputs] = useState({ input1: '', input2: '' });
    const [numbers, setNumbers] = useState({ number1: 0, number2: 0 });
    const [sum, setSum] = useState(undefined);

    const handleChange = (event) =>
        setInputs((prevState) => ({
            ...prevState,
            [event.target.name]: event.target.value,
        }));

    // const handleSum = () => {
    //     const { input1, input2 } = inputs;
    //     setSum(Number(input1) + Number(input2));
    //     setNumbers({
    //         number1: input1 === null || input1 === "" ? 0 : input1, //missing to insert value in any input box
    //         number2: input2 === null || input2 === "" ? 0 : input2,
    //     });
    // };
    function handleSum() {
        const { input1, input2 } = inputs;
        setSum(Number(input1) + Number(input2));
    };
    console.log(handleSum, "handleSum");

    //Setting AutoFocus for first input box
    const firstInputRef = React.useRef(null);
    React.useEffect(() => {
        firstInputRef.current.focus();
    }, []);

    const handleSubmit = (event) => {
        event.preventDefault();
        setInputs({ input1: "", input2: "" });
        firstInputRef.current.focus(); //make it focus again onSubmit button
    };

    return (
        <div className="container">
            <div className="header">
                <h1>6Unit Test of input elements</h1>
            </div>

            <form onSubmit={handleSubmit}>
                <div>
                    <label data-testid="myInput1" htmlFor="input1">
                        First Integer:
                    </label>
                    <input
                        type="number"
                        id="input1"
                        name="input1"
                        // autoFocus="true"
                        ref={firstInputRef}
                        value={inputs.input1}
                        onChange={handleChange}
                    ></input>
                </div>
                <div>
                    <label data-testid="myInput2" htmlFor="input2">
                        Second Integer:
                    </label>
                    <input
                        type="number"
                        id="input2"
                        name="input2"
                        value={inputs.input2}
                        onChange={handleChange}
                    ></input>
                </div>
                <div>
                    <button data-testid="myBtn" type="submit" onClick={handleSum}>
                        Sum
                    </button>
                </div>
            </form>
            <div>
                {sum !== undefined && (
                    <h3>
                        Result : {numbers.number1} + {numbers.number2} = {sum}
                    </h3>
                )}
            </div>
        </div>
    );
};

export default Sum;
