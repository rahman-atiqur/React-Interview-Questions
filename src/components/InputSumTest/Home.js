import React from 'react'
import Sum from './Sum'
import "./Style.css"

const Home = () => {
  return (
    <div>
        <Sum />
    </div>
  )
}

export default Home
