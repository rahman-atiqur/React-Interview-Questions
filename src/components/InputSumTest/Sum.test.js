import React from "react";
// import { ReactDOM } from "react-dom";
import Sum from "./Sum";
import Home from "./Home";


import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

// if("Testing Sum component for button",()=>{

//     const {getByTestId} = render(<Sum label="save" />);
//     expect(screen.getByTestId("myBtn")).toHaveTextContent("save")

// });


// RTL

//Button Testing
it("Testing for button label by getByTestId", () => {
        const { getByTestId } = render(<Home label="Sum" />);
        expect(getByTestId("myBtn")).toHaveTextContent("Sum");

    });


it("Testing for button label by getByText",() => {
        const { getByText } = render(<Home label="Sum" />);
        expect(getByText("Sum")).toBeInTheDocument();
    })

// Snapshot testing
it("Snapshot testing for button label",()=>{
    const tree=renderer.create(<Home label="Sum" />).toJSON;
    expect(tree).toMatchSnapshot();
})

//First input label testing
it("Testing for first input label by getByTestId", () => {
    const { getByTestId } = render(<Sum label="First Integer:" />);
    expect(getByTestId("myInput1")).toHaveTextContent("First Integer:");

});

//Second input label testing
it("Testing for second input label by getByTestId", () => {
    const { getByTestId } = render(<Sum label="Second Integer:" />);
    expect(getByTestId("myInput2")).toHaveTextContent("Second Integer:");

});




