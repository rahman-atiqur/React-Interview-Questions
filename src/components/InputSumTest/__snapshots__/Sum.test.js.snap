// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for button  1`] = `[Function]`;

exports[`Snapshot testing for button 1`] = `[Function]`;

exports[`Snapshot testing for button label 1`] = `[Function]`;

exports[`Snapshot testing for button label22 1`] = `[Function]`;
